Kickstarter-iOS/Views/Controllers/ProjectPageViewController.swift

```diff
@@ -0,0 +1,415 @@
+   Maybe check `BackerDashboardViewController`'s pageViewController for in-app examples on how to do this.
+   */
+  private var contentController: ProjectPamphletContentViewController?
+
+  // MARK: User session state properties
+
+  private var sessionEndedObserver: Any?
+  private var sessionStartedObserver: Any?
+
+  public var messageBannerViewController: MessageBannerViewController?
+
+  private lazy var navigationShareButton: UIBarButtonItem = {
+    let contentView = UIButton()
+      |> shareButtonStyle
+      |> UIButton.lens.imageEdgeInsets .~ UIEdgeInsets(left: -NavigationButtonSizes.spacing.rawValue)
+      |> UIButton.lens.accessibilityLabel %~ { _ in Strings.dashboard_accessibility_label_share_project() }
+
+    contentView.addTarget(
+      self,
+      action: #selector(ProjectPageViewController.shareButtonTapped),
+      for: .touchUpInside
+    )
+
+    let barButtonItem = UIBarButtonItem(customView: contentView)
+
+    return barButtonItem
+  }()
+
+  private lazy var navigationCloseButton: UIBarButtonItem = {
+    let contentView = UIButton(type: .custom)
+      |> UIButton.lens.title(for: .normal) .~ nil
+      |> UIButton.lens.image(for: .normal) .~ image(named: "icon--cross")
+      |> UIButton.lens.tintColor .~ .ksr_support_700
+      |> UIButton.lens.accessibilityLabel %~ { _ in Strings.accessibility_projects_buttons_close() }
+      |> UIButton.lens.accessibilityHint %~ { _ in Strings.Closes_project() }
+
+    contentView.addTarget(
+      self,
+      action: #selector(ProjectPageViewController.closeButtonTapped),
+      for: .touchUpInside
+    )
+
+    let barButtonItem = UIBarButtonItem(customView: contentView)
+
+    return barButtonItem
+  }()
+
+  private lazy var navigationSaveButton: UIBarButtonItem = {
+    let contentView = UIButton()
+      |> saveButtonStyle
+      |> UIButton.lens.accessibilityLabel %~ { _ in Strings.Toggle_saving_this_project() }
+
+    contentView
+      .addTarget(self, action: #selector(ProjectPageViewController.saveButtonTapped(_:)), for: .touchUpInside)
+    contentView
+      .addTarget(self, action: #selector(ProjectPageViewController.saveButtonPressed), for: .touchDown)
+
+    let barButtonItem = UIBarButtonItem(customView: contentView)
+
+    return barButtonItem
+  }()
+
+  private let pledgeCTAContainerView: PledgeCTAContainerView = {
+    PledgeCTAContainerView(frame: .zero) |> \.translatesAutoresizingMaskIntoConstraints .~ false
+  }()
+
+  public static func configuredWith(
+    projectOrParam: Either<Project, Param>,
+    refTag: RefTag?
+  ) -> ProjectPageViewController {
+    let vc = ProjectPageViewController.instantiate()
+
+    vc.viewModel.inputs.configureWith(projectOrParam: projectOrParam, refTag: refTag)
+
+    return vc
+  }
+
+  public override func viewDidLoad() {
+    super.viewDidLoad()
+
+    self.configurePledgeCTAContainerView()
+
+    /** FIXME:  - https://kickstarter.atlassian.net/browse/NTV-195
```

Comment:
Assuming this is where we plug in our new page view controller and the first screen that loads by default?

Reply:
Yep!